pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -78,8 +79,8 @@ func NewController(
 	virtualServiceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("ClusterIngress")),
 		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    impl.EnqueueControllerOf,
-			UpdateFunc: controller.PassNew(impl.EnqueueControllerOf),
+			AddFunc:    c.enqueueOwnerIngress(impl),
+			UpdateFunc: controller.PassNew(c.enqueueOwnerIngress(impl)),
```

Comment:
I think I mistook this for the `Route` -> `ClusterIngress` relationship, which can't use `OwnerReference`.

Since `ClusterIngress` isn't namespaced, I believe my prior comment was false here (but accurate for when we move to `ClusterIngress` in the `Route` controller).

I apologize for the randomization, but you should be able to revert this change.

Reply:
Yup, it can work from a functional standpoint, as we don't need namespace info for cluster-scoped resources.
But seeing it from a design perspective, the `OwnerReference ` is supposed for "resources of same namespace", but not for cluster-scoped resources.
So I'm thinking if we should just get rid of this, to avoid possible semantic changes in the future.

Reply:
That way, in `EnqueueControllerOf`, the key enqueued is indeed "virtualservice namespace+ingress name“: https://github.com/knative/pkg/blob/master/controller/controller.go#L129

And in reconciler we just throw away the namespace to make it work, I think the behavior a little "opportunistic", WDYT?

Reply:
I'm happy to leave this, if you are.  We could probably use a generalized `impl.EnqueueLabvelOf(key)` method in `knative/pkg/controller` to generalize this pattern a bit (I've now seen it a handful of places).  Not for this PR.

Reply:
> We could probably use a generalized impl.EnqueueLabvelOf(key) method in knative/pkg/controller to generalize this pattern a bit (I've now seen it a handful of places)

That makes sense, we'll also need the kind of func in route->ingress.
I'll keep this in mind, and make it a cleanup after all the things are done